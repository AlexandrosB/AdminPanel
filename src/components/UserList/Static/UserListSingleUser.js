import React, { memo } from 'react';
import PropTypes from 'prop-types';

const UserListSingleUser = (props) => {
    const {
            userId, userPhoto, userName, userEmail, selectUser, selectedUserId
        } = props,
        singleUserClass = (userId === selectedUserId) ? 'singleUser selected' : 'singleUser';
    return (
        <li className={ singleUserClass } onClick={ () => selectUser(userId) }>
            <span className='singleUserPhoto'>
                <img src={ userPhoto } alt='User' />
            </span>
            <span className='singleUserExtra'>
                <div className='singleUserName'>
                    { userName }
                </div>
                <div className='singleUserEmail'>
                    { userEmail }
                </div>
            </span>
        </li>
    );
};

UserListSingleUser.propTypes = {
    userId: PropTypes.string.isRequired,
    userPhoto: PropTypes.string.isRequired,
    userName: PropTypes.string.isRequired,
    userEmail: PropTypes.string.isRequired,
    selectUser: PropTypes.func.isRequired,
    selectedUserId: PropTypes.string.isRequired
};

export default memo(UserListSingleUser);
