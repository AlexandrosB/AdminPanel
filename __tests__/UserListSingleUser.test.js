import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import UserListSingleUser from '../src/components/UserList/Static/UserListSingleUser';

describe('<UserListSingleUser />', () => {
    const mockFnCb = jest.fn();
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<UserListSingleUser
            userId='dasdasdadas'
            userPhoto='random.url..com'
            userName='Me'
            userEmail='mymonkey@email.com'
            selectUser={ mockFnCb }
            selectedUserId='qwertytrewq'
        />);
    });

    it('should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        wrapper;
    });

    it('renders a list list item', () => {
        expect(wrapper.find('li.singleUser').exists()).toBe(true);
    });

    it('check the type of props', () => {
        const props = {
                userId: 'dasdasdadas',
                userPhoto: 'random.url.com',
                userName: 'Me',
                userEmail: 'mymonkey@email.com',
                selectedUserId: 'qwertytrewq'
            },
            mountWrap = mount(<UserListSingleUser { ...props } />);
        expect(mountWrap.prop('userId')).toBeString();
        expect(mountWrap.prop('userPhoto')).toBeString();
        expect(mountWrap.prop('userName')).toBeString();
        expect(mountWrap.prop('userEmail')).toBeString();
        expect(mountWrap.prop('selectedUserId')).toBeString();
    });
});
