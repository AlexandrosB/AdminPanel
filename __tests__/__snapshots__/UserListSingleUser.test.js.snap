// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserListSingleUser /> should match the snapshot 1`] = `
<li
  className="singleUser"
  onClick={[Function]}
>
  <span
    className="singleUserPhoto"
  >
    <img
      alt="User"
      src="random.url..com"
    />
  </span>
  <span
    className="singleUserExtra"
  >
    <div
      className="singleUserName"
    >
      Me
    </div>
    <div
      className="singleUserEmail"
    >
      mymonkey@email.com
    </div>
  </span>
</li>
`;
